core/src/main/java/ai/timefold/solver/core/impl/localsearch/decider/acceptor/ReconfigurableAcceptor.java

```diff
@@ -0,0 +1,70 @@
+package ai.timefold.solver.core.impl.localsearch.decider.acceptor;
+
+import ai.timefold.solver.core.api.score.Score;
+import ai.timefold.solver.core.impl.localsearch.decider.acceptor.restart.RestartStrategy;
+import ai.timefold.solver.core.impl.localsearch.scope.LocalSearchMoveScope;
+import ai.timefold.solver.core.impl.localsearch.scope.LocalSearchPhaseScope;
+import ai.timefold.solver.core.impl.localsearch.scope.LocalSearchStepScope;
+import ai.timefold.solver.core.impl.solver.scope.SolverScope;
+
+/**
+ * Base class designed to analyze whether the solving process needs to be restarted.
+ * Additionally, it also calls a reconfiguration logic as a result of restarting the solving process.
+ */
+public abstract class ReconfigurableAcceptor<Solution_> extends AbstractAcceptor<Solution_> {
+
+    private final RestartStrategy<Solution_> restartStrategy;
+
+    protected ReconfigurableAcceptor(RestartStrategy<Solution_> restartStrategy) {
+        this.restartStrategy = restartStrategy;
+    }
+
+    @Override
+    public void solvingStarted(SolverScope<Solution_> solverScope) {
+        super.solvingStarted(solverScope);
+        restartStrategy.solvingStarted(solverScope);
+    }
+
+    @Override
+    public void phaseStarted(LocalSearchPhaseScope<Solution_> phaseScope) {
+        super.phaseStarted(phaseScope);
+        restartStrategy.phaseStarted(phaseScope);
+    }
+
+    @Override
+    public void phaseEnded(LocalSearchPhaseScope<Solution_> phaseScope) {
+        super.phaseEnded(phaseScope);
+        restartStrategy.phaseEnded(phaseScope);
+    }
+
+    @Override
+    public void stepStarted(LocalSearchStepScope<Solution_> stepScope) {
+        super.stepStarted(stepScope);
+        restartStrategy.stepStarted(stepScope);
+    }
+
+    @Override
+    public void stepEnded(LocalSearchStepScope<Solution_> stepScope) {
+        super.stepEnded(stepScope);
+        restartStrategy.stepEnded(stepScope);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public boolean isAccepted(LocalSearchMoveScope<Solution_> moveScope) {
+        if (restartStrategy.isTriggered(moveScope)) {
+            moveScope.getStepScope().getPhaseScope().triggerReconfiguration();
+            return true;
+        }
+        var accepted = evaluate(moveScope);
+        var improved = moveScope.getScore().compareTo(moveScope.getStepScope().getPhaseScope().getBestScore()) > 0;
+        if (improved) {
+            restartStrategy.reset(moveScope);
+        }
+        return accepted;
+    }
+
+    protected abstract boolean evaluate(LocalSearchMoveScope<Solution_> moveScope);
+
+    protected abstract <Score_ extends Score<Score_>> void reset(Score_ score);
```

Comment:
Reset what? Why?
Arguably this method needs a better name, and/or Javadoc.

Reply:
I realized we don't need this method.